benchmarks/be_bm_utils.erl

```diff
@@ -0,0 +1,93 @@
+-module(be_bm_utils).
+
+%%
+%% Util functions for benchmarking
+%%
+
+%% API
+-export([
+  better_percentage/1,
+
+  file_exists/1,
+
+  betree_allocations/0,
+  betree_allocations_diff/2,
+
+  diff_tuple/2,
+
+  write_terms/2
+]).
+
+%% Consider a sorted list of numbers where
+%%  - negative numbers indicates a 'better';
+%%  - positive number indicates a 'worse'.
+%% better_percentage finds how many 'betters' are in the list
+%%  expressed as percentage.
+better_percentage([]) ->
+  not_applicable;
+better_percentage([Min | _] = NsSorted) ->
+  case Min >= 0 of
+    true -> 0;
+    _ ->
+      Max = lists:last(NsSorted),
+      case Max =< 0 of
+        true -> 100;
+        _ ->
+          Enumed = lists:enumerate(NsSorted),
+          case lists:search(fun ({_N, Rel}) -> Rel >= 0 end, Enumed) of
+            false -> unknown;
+            {value, {N0, _Rel0}} ->
+              Total = length(NsSorted),
+              floor((N0/Total) * 100)
+          end
+      end
+  end.
+
+file_exists(FileName) ->
+  case file:open(FileName, [read]) of
+    {ok, Handle} ->
+      file:close(Handle),
+      true;
+    _ -> false
+  end.
```

Comment:
maybe we can use filelib:is_file here ?

Reply:
**filelib:is_file** does not make a distinction between a file or a directory. Probably that was a reason why I've chosen **file:open/close**.
I understand, **file:open** is an expensive operation.
In the benchmarking process it is used only at the beginning of the process just to ensure that input files are in place.
Would prefer to leave it as is if you don't insist.

Reply:
Ok. Thanks.